anomaly_detector/storage/storage_proxy.py

```diff
@@ -3,15 +3,23 @@
 from anomaly_detector.storage.storage_catalog import StorageCatalog
 from anomaly_detector.storage.storage_source import StorageSource
 from anomaly_detector.storage.storage_sink import StorageSink
+from anomaly_detector.exception.exceptions import InvalidStorageProxyConfigException
 
 
 class StorageProxy(StorageSource, StorageSink):
     """Storage Proxy for facilitating communication with backend."""
 
+    SUFFIX_SOURCE = ".source"
+    SUFFIX_SINK = ".sink"
+
     def __init__(self, config: Configuration):
         """Create storage data source and sinks to talk to storage backend."""
-        self.source = StorageCatalog(config=config, storage_api=config.STORAGE_DATASOURCE).get_storage_api()
-        self.sink = StorageCatalog(config=config, storage_api=config.STORAGE_DATASINK).get_storage_api()
+        if config.STORAGE_DATASOURCE.endswith(self.SUFFIX_SOURCE) or config.STORAGE_DATASINK.endswith(self.SUFFIX_SINK):
```

Comment:
Just wondering - why test for only ".sink" ".source" ? A user could put in ".foo" or ".bar" and pass this right? Would it not be better to check if it matches the existing supported source/sinks instead? e.g. something like "config.STORAGE_DATASINK in ['es', 'local', ... ec.] ?

Reply:
We are already checking lower down in the stack. So I'll remove this.